dask/dataframe/partitionquantiles.py

```diff
@@ -401,19 +401,22 @@ def percentiles_summary(df, num_old, num_new, upsample, state):
         Scale factor to increase the number of percentiles calculated in
         each partition.  Use to improve accuracy.
     """
-    from dask.array.percentile import _percentile
+    from dask.dataframe.dispatch import _percentile
 
     length = len(df)
     if length == 0:
         return ()
     random_state = np.random.RandomState(state)
     qs = sample_percentiles(num_old, num_new, length, upsample, random_state)
-    data = df.values
+    data = df
     interpolation = "linear"
+
     if is_categorical_dtype(data):
-        data = data.codes
+        data = data.cat.codes
         interpolation = "nearest"
-    elif np.issubdtype(data.dtype, np.integer) and not is_cupy_type(data):
+    elif isinstance(data.dtype, pd.core.dtypes.dtypes.DatetimeTZDtype) or np.issubdtype(
+        data.dtype, np.integer
+    ):
```

Comment:
Does the new condition also cover the case for which the `not is_cupy_type(data)` check was here?

Reply:
Had to remove this check as cupy now supports `interpolation = "nearest" & "linear"`